src/main/java/io/specto/hoverfly/junit/verification/InOrderVerification.java

```diff
@@ -0,0 +1,54 @@
+package io.specto.hoverfly.junit.verification;
+
+import io.specto.hoverfly.junit.api.HoverflyClient;
+import io.specto.hoverfly.junit.core.model.JournalEntry;
+import io.specto.hoverfly.junit.dsl.RequestMatcherBuilder;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static io.specto.hoverfly.junit.verification.HoverflyVerifications.times;
+
+public class InOrderVerification {
+
+
+    private HoverflyClient hoverflyClient;
+    private List<RequestMatcherBuilder> requestMatchers = new ArrayList<>();
+
+    public InOrderVerification(HoverflyClient hoverflyClient, RequestMatcherBuilder... requestMatchers) {
+        this.hoverflyClient = hoverflyClient;
+        this.requestMatchers.addAll(Arrays.asList(requestMatchers));
+    }
+
+    public void verify() {
```

Comment:
I don't think this would work. If you want to verify this:

```
verifyInOrder(matcherA, matcherB, matcherA)
```

Then it will always fail, because you only allow each element to appear once. Could you reproduce it with a test?

What I would do is, is for each request get the results from the journal and put them into an ordered set based on time. Duplicates will be the same matcher and time. Once you have finished, just loop the set and verify each element one by one.